File: mycollab-ui/src/main/java/com/esofthead/mycollab/vaadin/resources/file/FileAssetsUtil.java
/**
 * This file is part of mycollab-ui.
 *
 * mycollab-ui is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-ui is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-ui.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.vaadin.resources.file;

import com.esofthead.mycollab.core.utils.MimeTypesUtil;
import com.vaadin.server.FontAwesome;

import java.util.HashMap;
import java.util.Map;

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
public class FileAssetsUtil {
    private static final Map<String, FontAwesome> extraIconMap;

    static {
        extraIconMap = new HashMap<>();
        extraIconMap.put("application/pdf", FontAwesome.FILE_PDF_O);
        extraIconMap.put("application/vnd.ms-excel", FontAwesome.FILE_EXCEL_O);
        extraIconMap.put("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", FontAwesome.FILE_EXCEL_O);
        extraIconMap.put("application/msword", FontAwesome.FILE_WORD_O);
        extraIconMap.put("application/vnd.openxmlformats-officedocument.wordprocessingml.document", FontAwesome.FILE_WORD_O);
        extraIconMap.put("application/vnd.ms-powerpoint", FontAwesome.FILE_POWERPOINT_O);
        extraIconMap.put("application/vnd.openxmlformats-officedocument.presentationml.presentation", FontAwesome.FILE_POWERPOINT_O);
        extraIconMap.put("application/zip", FontAwesome.FILE_ZIP_O);
        extraIconMap.put("application/x-rar-compressed", FontAwesome.FILE_ARCHIVE_O);
        extraIconMap.put("application/x-7z-compressed", FontAwesome.FILE_ARCHIVE_O);
        extraIconMap.put("application/java-archive", FontAwesome.FILE_ARCHIVE_O);
        extraIconMap.put("application/x-tar", FontAwesome.FILE_ARCHIVE_O);
    }

    public static FontAwesome getFileIconResource(String fileName) {
        String mimeType = MimeTypesUtil.detectMimeType(fileName);
        if (MimeTypesUtil.isImage(mimeType)) {
            return FontAwesome.FILE_IMAGE_O;
        } else if (MimeTypesUtil.isAudio(mimeType)) {
            return FontAwesome.FILE_AUDIO_O;
        } else if (MimeTypesUtil.isVideo(mimeType)) {
            return FontAwesome.FILE_VIDEO_O;
        } else if (MimeTypesUtil.isText(mimeType)) {
            return FontAwesome.FILE_TEXT_O;
        }
        FontAwesome icon = extraIconMap.get(mimeType);
        return (icon == null) ? FontAwesome.FILE_O : icon;
    }
}

File: mycollab-web/src/main/java/com/mycollab/vaadin/resources/file/FileAssetsUtil.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.vaadin.resources.file

import com.mycollab.core.utils.MimeTypesUtil
import com.vaadin.icons.VaadinIcons

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object FileAssetsUtil {
    private val extraIconMap = mapOf("application/pdf" to VaadinIcons.FILE_TEXT,
            "application/vnd.ms-excel" to VaadinIcons.FILE_CODE,
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet" to VaadinIcons.FILE_CODE,
            "application/msword" to VaadinIcons.FILE_CODE,
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document" to VaadinIcons.FILE_CODE,
            "application/vnd.ms-powerpoint" to VaadinIcons.FILE_PRESENTATION,
            "application/vnd.openxmlformats-officedocument.presentationml.presentation" to VaadinIcons.FILE_PRESENTATION,
            "application/zip" to VaadinIcons.FILE_ZIP,
            "application/x-rar-compressed" to VaadinIcons.FILE_ZIP,
            "application/x-7z-compressed" to VaadinIcons.FILE_ZIP,
            "application/java-archive" to VaadinIcons.FILE_ZIP,
            "application/x-tar" to VaadinIcons.FILE_ZIP)

    @JvmStatic fun getFileIconResource(fileName: String): VaadinIcons {
        val mimeType: String = MimeTypesUtil.detectMimeType(fileName)
        return when {
            MimeTypesUtil.isImage(mimeType) -> VaadinIcons.FILE_PICTURE
            MimeTypesUtil.isAudio(mimeType) -> VaadinIcons.FILE
            MimeTypesUtil.isVideo(mimeType) -> VaadinIcons.FILE_MOVIE
            MimeTypesUtil.isText(mimeType) -> VaadinIcons.FILE_TEXT_O
            else -> extraIconMap[mimeType] ?: VaadinIcons.FILE_O
        }
    }
}